js/scripts/JSRootPainter.v7.js

```diff
@@ -4262,6 +4262,23 @@
       return col;
    }
 
+   TCanvasPainter.prototype.GetNewOpt = function(opts, name) {
+      if (!opts || !opts.fHolderIO || !name) return;
+
+      var map = opts.fHolderIO.fAttrNameVals;
+      if (!map || !map.length) return;
+
+      for (var i=0; i<map.length; ++i)
```

Comment:
Here, a JS map https://stackoverflow.com/questions/1208222/how-to-do-associative-array-hashing-in-javascript could make things *much* faster...

Reply:
Yes, but this requires creation of such JSON object already on C++ side.
In principle - possible, but not principal right now.
Much more important - play-through all scenarios with CSS usage

Reply:
Here is PR #3701 which optionally enable such feature for selected C++ objects/data members. Potentially can be enabled for attributes storage in the future